src/AbstractEnum.php

```diff
@@ -70,6 +81,29 @@
     }
 
     /**
+     * @param callable|null $callback           A callable function compatible with array_map
+     * @param string|null   $namespaceSeparator Choose which character should replace namespaces separation.
+     *                                          Example: With Foo\BarMagic enum class with '.' separator,
+     *                                          it will be converted to foo.bar_magic.YOUR_KEY
+     *
+     * @return array|string[]
+     */
+    final public static function getClassPrefixedKeys($callback = null, $namespaceSeparator = self::DEFAULT_NAMESPACE_SEPARATOR)
+    {
+        $classKey = str_replace('\\', $namespaceSeparator, Inflector::tableize(static::class));
+
+        $keys = static::getKeys(function ($key) use ($namespaceSeparator, $classKey) {
+            return $classKey.$namespaceSeparator.$key;
+        });
+
+        if (null !== $callback) {
```

Comment:
I mean "use the same thing everywhere", how is that not normalization ? :confused: